engine/access/state_stream/backend/backend_executiondata2_test.go

```diff
@@ -0,0 +1,1065 @@
+						return nil, assert.AnError
+					}).
+					Once()
+			},
+		},
+	}
+
+	s.headers.
+		On("BlockIDByHeight", mock.Anything).
+		Return(func(height uint64) (flow.Identifier, error) {
+			block, ok := s.blocksHeightToBlockMap[height]
+			if !ok {
+				return flow.ZeroID, storage.ErrNotFound
+			}
+			return block.ID(), nil
+		})
+
+	s.executionDataTracker.
+		On("GetHighestAvailableFinalizedHeight").
+		Return(s.blocks[len(s.blocks)-1].Height)
+
+	executionDataReader := osyncmock.NewBlockExecutionDataReader(s.T())
+	s.executionDataSnapshot.
+		On("BlockExecutionData").
+		Return(executionDataReader)
+
+	s.executionStateCache.
+		On("Snapshot", mock.Anything).
+		Return(s.executionDataSnapshot, nil)
+
+	executionDataReader.
+		On("ByBlockID", mock.Anything, mock.Anything).
+		Return(func(ctx context.Context, blockID flow.Identifier) (*execution_data.BlockExecutionDataEntity, error) {
+			ed, ok := s.blockIDToExecutionDataMap[blockID]
+			if !ok {
+				return nil, storage.ErrNotFound
+			}
+
+			return &execution_data.BlockExecutionDataEntity{
+				BlockExecutionData: ed,
+				ExecutionDataID:    unittest.IdentifierFixture(),
+			}, nil
+		})
+
+	s.executionDataTracker.
+		On("GetStartHeightFromBlockID", mock.AnythingOfType("flow.Identifier")).
+		Return(
+			func(ID flow.Identifier) (uint64, error) {
+				return s.executionDataTrackerReal.GetStartHeightFromBlockID(ID)
+			},
+			nil,
+		).
+		Times(len(tests))
+
+	backend := NewExecutionDataBackend(
+		s.log,
+		s.state,
+		s.headers,
+		s.subscriptionFactory,
+		s.executionDataTracker,
+		s.executionResultProvider,
+		s.executionStateCache,
+	)
+
+	for _, test := range tests {
+		s.Run(test.name, func() {
+			test.mockState()
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			currentHeight := s.sporkRootBlock.Height
+			sub := backend.SubscribeExecutionDataFromStartBlockID(ctx, s.sporkRootBlock.ID(), s.criteria)
+
+			for value := range sub.Channel() {
+				actualExecutionData, ok := value.(*ExecutionDataResponse)
+				require.True(s.T(), ok, "expected *ExecutionDataResponse on the channel")
+				require.NotNil(s.T(), actualExecutionData.ExecutionData, "expected non-nil execution data")
+
+				block := s.blocksHeightToBlockMap[currentHeight]
+				expectedExecutionData := s.blockIDToExecutionDataMap[block.ID()]
+				require.Equal(s.T(), expectedExecutionData, actualExecutionData.ExecutionData)
+
+				currentHeight += 1
+			}
+
+			require.ErrorIs(s.T(), sub.Err(), test.expectedErr)
+		})
+	}
+}
+
+// TestExecutionDataProviderIgnorableErrors verifies that the data provider handles temporary or
+// ignorable errors correctly. It ensures that the provider retries or waits when encountering
+// errors like missing required executors, rather than terminating the subscription immediately,
+// until the context is canceled.
+func (s *BackendExecutionDataSuite2) TestExecutionDataProviderIgnorableErrors() {
```

Comment:
I don't think "ignorable" is the right framing. how about
```suggestion
func (s *BackendExecutionDataSuite2) TestExecutionDataProviderCriteriaNotMet() {
```